src/raftstore/store/snap_io_limiter.rs

```diff
@@ -0,0 +1,181 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+use std::io::{Result, Write};
+
+use rocksdb::RateLimiter;
+
+use server::config::{DEFAULT_SNAP_MAX_BYTES_PER_SEC, DEFAULT_SNAP_MAX_BYTES_PER_TIME,
+                     DEFAULT_SNAP_MIN_BYTES_PER_TIME};
+
+const REFILL_PERIOD: i64 = 100 * 1000;
+const FARENESS: i32 = 10;
+
+pub struct SnapshotIOLimiter {
```

Comment:
Seems this limiter is not specified to snapshot, you can create a generic limiter type as a utility.

Reply:
Sure, i can change the name.